database_benchmarking_suite/src/main/scala/uk/ac/ebi/eva/benchmarking_suite/mongodb/MongoDBReadSampler.scala

```diff
@@ -19,17 +22,23 @@ class MongoDBReadSampler() extends AbstractSampler {
         mongoDBTestParams = JMeterUtils.getJMeterProperties.get("connectionParams").asInstanceOf[MongoDBConnectionParams]
         val numReadsPerThread = this.getPropertyAsInt("numOpsPerThread")
         val threadNum = this.getThreadContext.getThreadNum
-        randomNumGen = new scala.util.Random(threadNum)
+        //Use thread number and timestamp to vary the random seed across multiple runs for a same thread choice
+        randomNumGen = new scala.util.Random(threadNum + System.currentTimeMillis())
         (1 to numReadsPerThread).foreach(_ => readData())
       })
   }
 
   def readData(): Unit = {
-    val chromosome = randomNumGen.nextInt(16)
-    val startPos = randomNumGen.nextInt(1e9.toInt/16)
-    mongoDBTestParams.mongoCollection.find(
-      and(equal("chromosome", chromosome),
+    val chromosome = randomNumGen.nextInt(32)
+    val startPos = randomNumGen.nextInt(1e8.toInt/32)
+    val findResults = mongoDBTestParams.mongoCollection.find(
+      and(equal("chromosome", chromosome.toString),
         gt("start_pos", startPos),
-        lte("start_pos", startPos + blockReadSize))).foreach(doc => doc.get("start_pos")) //Force document retrieval by getting one attribute
+        lte("start_pos", startPos + blockReadSize))).maxTime(10.minutes)
```

Comment:
what happens if the 10 minutes pass? is the query cancelled with error?

Reply:
Query times out with Timeout error for that particular thread and the error appears in the result log (with error codes marked in *.jtl files. see those attached to the ticket). I haven't observed any errors in the benchmarking that I ran. Probably because 10 minutes is plenty of time to retrieve 100 rows from a query that only references fully indexed attributes.